### PR TITLE
Add revision options to API for git compiling (#19)

### PR DESCRIPTION
**Pull request by NonlyNerd**

### CODE REVIEW

File: lib/DownloadManager.js [hunk 1]
```diff
@@ -85,8 +86,9 @@ class DownloadManager {
             }
             var fulfill;
             var promise = new Promise(x => fulfill = x);
-            logger.info(`git clone ${url} ${folderPath}`);
-            exec('git', ['clone', '--depth', '1', url, folderPath], {timeout: utils.minutes(2)}, (err, stdout, stderr) => {
+            var args = ['clone', '--depth', '1', '-b', revision, url, folderPath];
```
Comment (aslushnikov) [inline lib/DownloadManager.js]: @NonlyNerd this doesn't seem to work for me:

```
$ git clone https://github.com/aslushnikov/latex-online --depth 1 -b 18b2131a29b1946e0180e2f58493eef988b2950d
Cloning into 'latex-online'...
warning: Could not find remote branch 18b2131a29b1946e0180e2f58493eef988b2950d to clone.
fatal: Remote branch 18b2131a29b1946e0180e2f58493eef988b2950d not found in upstream origin
```

Comment (NonlyNerd) [reply lib/DownloadManager.js]: My bad. I forgot the shallow thing and the handling of git-clone for revision ( I mixed up checkout and clone). For checking with a commit identifier, we have to work with a full clone, not a shallow one. I've updated the README to not disrupt users on the usage.

**Comment by aslushnikov** [comment]
Looks like there's been no progress here - closing for now. Feel free to re-submit if this is still relevant.

**Comment by rounakdatta** [comment]
@aslushnikov @NonlyNerd can we re-open this? I'd love to finish the remaining work on it :)

**Comment by aslushnikov** [comment]
@rounakdatta feel free to send another PR!

**Pull request closed without merging**